### PR TITLE
POST booking api and GET userlist api

### PR DESCRIPTION
**Pull request by Krutikaa04**
A new bookingSchema was created under model.
POST booking api and GET userlist api are created.
A collection is created under database named bookings.

### CODE REVIEW

**Comment by Krutikaa04** [comment]
made some more changes as follows : 
made a POST registration api
changed the login api from POST to GET
added the autheticate credentials feature in login api

File: app.js [hunk 1]
```diff
@@ -24,41 +25,76 @@ mongoose.connect(db, {
...
+    const checkUser = await Users.findOne({ email: email, password: password});
+    if(!checkUser){
+      res.status(400).json({error: "Invalid Credentials"});
+    }
+    else{
+      token = jwt.sign({_id: this._id, email: this.email}, process.env.SECRET_TOKEN, {expiresIn: "24h"});
+      
+      res.status(200).json({
+        message: "You logged in successfully ",
+        token: token,
```
Comment (shahkushal38) [inline app.js]: Along with token also please send the user email and username

File: app.js [hunk 2]
```diff
@@ -82,6 +118,50 @@ app.get('/profile', async (req, res)=>{
 
 
 
+//booking api started
+
+
+app.post('/booking', async (req, res)=>{
+  const {username, email, branch, date, deskNo} = req.body;
+
+   await Users.findOne({ email: email}).then((userExist)=>{
```
Comment (shahkushal38) [inline app.js]: Find by email and username both

File: app.js [hunk 3]
```diff
@@ -82,6 +118,50 @@ app.get('/profile', async (req, res)=>{
...
+  
+    if(userExist){
+      const bookings = new Booking({username, email, branch, date, deskNo});
```
Comment (shahkushal38) [inline app.js]: Here you are not handling the case that if the desk is already booked for that particular date and time, then you should give and error response that desk is already booked

File: model/bookingSchema.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    branch: {
+        type: String,
+        required: true
+    },
+    date:{
+        type: Date,
+        default: Date.now(),
+        required:true
+    },
```
Comment (shahkushal38) [inline model/bookingSchema.js:20]: Add time i.e. the slot in that particular date along with Date for booking

File: app.js (hunk 3 above)
Comment (shahkushal38) [inline app.js]: Date and slot time needs to be saved in booking table

**Review by shahkushal38 — CHANGES_REQUESTED** [review]
Reviewed with comments

**Review by shahkushal38 — APPROVED** [review]
approved

**Review by Abhiru0116 — APPROVED** [review]
approved

**Review by Abhiru0116 — APPROVED** [review]
approved!

**Pull request merged**